routes/todos.js

```diff
@@ -1,68 +1,93 @@
-const { v4: uuidv4 } = require('uuid');
 const express = require('express');
 const router = express.Router();
-const moment = require('moment');
-
-
-const todos = [{"id": "1", "name": "test", "date": "03/04/2020"}, {"id": "2", "name": "test2", "date": "03/04/2020"}]
+const pgp = require('pg-promise')();
+const cn = {
+  host: process.env.DB_HOST,
+  port: process.env.DB_PORT,
+  database: process.env.DB_DATABASE,
+  user: process.env.DB_USER,
+  password: process.env.DB_PASSWORD,
+}
+const db = pgp(cn);
 
-router.get('/', function(req, res) {
-  res.send(todos);
+/**
+ * get a list of all the todos
+ */
+router.get('/', (req, res) => {
+  db.many('SELECT * FROM todo ORDER BY id')
+    .then(function (data) {
+      res.send(data);
+    })
+    .catch(function (error) {
+      res.status(404).send();
+    })
 });
 
+/**
+ * get one todo by id
+ */
 router.get('/:id', (req, res) => {
-  const todo = todos.find(element => element.id === req.params.id);
-  if (todo) {
-    res.status(200).send(todo);
-  } else {
-    res.status(404).send();
-  }
+  db.one('SELECT * FROM todo WHERE id = $1', req.params.id)
+    .then(function (data) {
+      res.send(data);
+    })
+    .catch(function (error) {
+      res.status(404).send();
+    })
 });
 
+/**
+ * create a new todo
+ */
 router.post('/', (req, res) => {
   const name = req.body.name;
 
   if (!name) {
     res.status(400).json({error: "Name can't be null"});
   } else {
-    const id = uuidv4();
-    const date = moment().format();
-
-   const todo = {id, name, date}
-
-   todos.push(todo);
-
-   res.send(todo);
-}
+    db.one('INSERT INTO todo(name) VALUES($1) RETURNING *', name) 
+    .then(function (data) {
+      res.status(200).send(data);
+    })
+    .catch(function (error) {
+      res.status(404).send(error);
+    })
+  }
 });
 
+/**
+ * modify the name of a todo by id
+ */
 router.put('/:id', (req, res) => {
   const name = req.body.name;
-  const id = req.params.id
+  const id = req.params.id;
 
   if (!name) {
     res.status(400).json({error: "Name can't be null"});
   } else {
-    const index = todos.findIndex(element => element.id === id);
-    if (index !== -1) {
-      todos[index].name = name;
-      res.send(todos[index]);
-    } else {
-      res.status(404).send();
-    }
+    db.one('UPDATE todo SET name = $1 WHERE id = $2 RETURNING *', [name, id]) 
+    .then(function (data) {
+      res.status(200).send(data);
+    })
+    .catch(function (error) {
+      res.status(404).send(error);
```

Comment:
In this case is more complicated, we should return a 404 if the todo doesn't exist and a 500 if there is an error. Here we return a 404 even if we have an issue with the database.